bindings/python/test/test_pandas.py

```diff
@@ -129,23 +138,34 @@ def _create_data(self):
             if k.__name__ not in ("ObjectId", "Decimal128")
         }
         schema = {k: v.to_pandas_dtype() for k, v in arrow_schema.items()}
+        schema["Int64"] = pd.Int64Dtype()
+        schema["int"] = pd.Int32Dtype()
         schema["str"] = "U8"
         schema["datetime"] = "datetime64[ns]"
 
         data = pd.DataFrame(
             data={
-                "Int64": [i for i in range(2)],
-                "float": [i for i in range(2)],
-                "int": [i for i in range(2)],
-                "datetime": [datetime.datetime(1970 + i, 1, 1) for i in range(2)],
-                "str": [f"a{i}" for i in range(2)],
-                "bool": [True, False],
+                "Int64": [i for i in range(2)] + [None],
+                "float": [i for i in range(2)] + [None],
+                "int": [i for i in range(2)] + [None],
+                "datetime": [datetime.datetime(1970 + i, 1, 1) for i in range(2)] + [None],
+                "str": [f"a{i}" for i in range(2)] + [None],
+                "bool": [True, False, None],
             }
         ).astype(schema)
         return arrow_schema, data
 
     def test_write_schema_validation(self):
         arrow_schema, data = self._create_data()
+
+        # Work around https://github.com/pandas-dev/pandas/issues/11453.
+        def new_replace(k):
+            if k.value < 1:
+                return datetime.datetime(1970, 1, 1)
+            return k.replace(tzinfo=None)
+
+        data["datetime"] = data.apply(lambda row: new_replace(row["datetime"]), axis=1)
```

Comment:
That pandas ticket seems to be about .time() not working on NA types but we don't use .time() anywhere. Can you explain?

Reply:
Ah, meant to link to https://github.com/pandas-dev/pandas/issues/16248.  Also added context.

Reply:
I'm still confused since this code is changing the data _before_ passing it to an pymongoarrow method. What happens if the user actually calls write() with a NA datetime? And why are we clamping the datetime to `datetime.datetime(1970, 1, 1)`?

Reply:
They would see the same error as https://jira.mongodb.org/browse/FREE-165786, for which this is a workaround.  I just chose a random valid date.

Reply:
Thanks, 2 follow up questions.
1) Can we change this to something like?:
```python
if isinstance(k, Nat):
   return None
return k
```

2) this looks like something we should ideally be working around in write() itself. Can you open a ticket for making write() encode NaT datetimes as BSON null?

Reply:
If we return `None` we get the original error, so I had to return the workaround datetime object.  I filed https://jira.mongodb.org/browse/ARROW-136.